scripts/build/createTag.sh

```diff
@@ -0,0 +1,239 @@
+  #
+  LINES=$(wc -l $SPEC_FILE | awk '{ print $1 }')
+
+  #
+  # 3. Get the number of lines in rpm/SPECS/contextBroker.spec after the insertion
+  #
+  typeset -i LAST_LINES
+  LAST_LINES=$LINES-$LINE
+
+  #
+  # 4. To a temporal file, add the four 'chunks'
+  #
+  head -$LINE $SPEC_FILE        >  /tmp/contextBroker.spec
+
+  echo -n '* '                  >> /tmp/contextBroker.spec
+  echo $dateLine                >> /tmp/contextBroker.spec
+
+  cat $changelog                                  >> /tmp/contextBroker.spec
+  echo                                            >> /tmp/contextBroker.spec
+
+  tail -$LAST_LINES $SPEC_FILE  >> /tmp/contextBroker.spec
+    
+  # 5. Replace using the temporal file
+  mv /tmp/contextBroker.spec $SPEC_FILE
+
+  rm $changelog; touch $changelog
+}
+
+
+
+# ------------------------------------------------------------------------------
+#
+# Argument parsing
+#
+files=off
+commit=off
+tag=off
+push=off
+
+changelog=CHANGES_NEXT_RELEASE
+
+mode=""
+
+while [ "$#" != 0 ]
+do
+  if   [ "$1" == "-u" ];            then usage 0;
+  elif [ "$1" == "--mode" ];        then mode="$2"; shift;
+  elif [ "$1" == "--changelog" ];   then changelog="$2"; shift;
+  else
+    echo $0: bad parameter/option: "'"${1}"'";
+    echo
+    usage 1
+  fi
+  shift
+done
+
+if [ "$mode" == "" ]; then
+  echo $0: missing mode
+  echo
+  usage 1
+fi
+
+if [ "$mode" == "files" ]; then
+  files=on  # not actually used, but included for "symetry"
+elif [ "$mode" == "commit" ]; then
+  commit=on  
+elif [ "$mode" == "tag" ]; then
+  commit=on
+  tag=on
+elif [ "$mode" == "push" ]; then
+  commit=on
+  tag=on
+  push=on
+else
+  echo $0: wrong mode: "'"${mode}"'";
+  echo
+  usage 1
+fi
+
+# Check we are in a release/X.Y.Z branch
+branch=$(git rev-parse --abbrev-ref HEAD)
+if [[ $branch != release/* ]]; then
+  echo $0: you are not in a release branch but in "'"${branch}"'"
+  exit 1
+fi
+
+
+# Get current tag
+currentTag=$(grep "define ORION_VERSION" $PROJECT_DIR/src/app/contextBroker/version.h  | awk -F\" '{ print $2 }')
+echo $0: current tag is: "'"${currentTag}"'"
+
+# Check tag structure
+X=$(echo $currentTag | awk -F '.' '{print $1}')
+Y=$(echo $currentTag | awk -F '.' '{print $2}')
+Z=$(echo $currentTag | awk -F '.' '{print $3}')
+checkValidInteger $X
+checkValidInteger $Y
+checkValidInteger $Z
+
+# Check branch match base version
+baseVersion=$X.$Y.0
+echo $0: base version is: "'"${baseVersion}"'"
+versionFromBranch=$(echo $branch | awk -F '/' '{print $2'})
+if [ "$baseVersion" != "$versionFromBranch" ]; then
+  echo $0: wrong base version in this release branch: "'"${versionFromBranch}"'"
+  exit 1
+fi
+
+# Check dockerfile has the right tag
+grep "ENV GIT_REV_ORION $currentTag" $PROJECT_DIR/docker/Dockerfile
+if [ "$?" != "0" ]; then
+  echo $0: GIT_REV_ORION does not use current tag in Dockerfile
+  exit 1
+fi
+
+# Calculate next tag
+Z=$((Z+1))
+nextTag=$X.$Y.$Z
+echo $0: next tag is: "'"${nextTag}"'"
```

Comment:
No, not really. Part of the normal output of the script, in order to report user how the things are going.